### ISSUE TITLE
Column title incorrect when joining via shared foreign key

### ISSUE THREAD

**Issue by rigoleto** [metabase/metabase#11452]
**Describe the bug**
I try to show this clearly in the screenshot below. I am joining table **A** to table **C** via **b_id** which both tables have. Metabase then shows `c.column_one` as **B -> Column One**. In the screenshot *grouping* area, all of the columns listed under the Account join actually belong to the Users+ table.

**To Reproduce**
1. Being a custom question
2. Join across tables via a 'shared' foreign key, as described above
3. Click 'Visualize' to see full table output.
4. Columns from the right join table are shown with an incorrect table name

**Expected behavior**
I would expect Account -> Email to show as Users+ -> Email since I'm using the email column on the Users+ table.

**Screenshots**
![metabase-join-table-name](https://user-images.githubusercontent.com/2439912/70279862-89935200-1774-11ea-8e7c-f0b4e8657b09.png)

**Information about your Metabase Installation:**
```
{
  "browser-info": {
    "language": "en-US",
    "platform": "Linux x86_64",
    "userAgent": "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:70.0) Gecko/20100101 Firefox/70.0",
    "vendor": ""
  },
  "system-info": {
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.3+7",
    "java.vendor": "AdoptOpenJDK",
    "java.vendor.url": "https://adoptopenjdk.net/",
    "java.version": "11.0.3",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.3+7",
    "os.name": "Linux",
    "os.version": "4.14.152-98.182.amzn1.x86_64",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "postgres"
    ],
    "hosting-env": "elastic-beanstalk",
    "application-database": "postgres",
    "run-mode": "prod",
    "version": {
      "date": "2019-11-19",
      "tag": "v0.33.6",
      "branch": "release-0.33.x",
      "hash": "be1e0e1"
    },
    "settings": {
      "report-timezone": "US/Mountain"
    }
  }
}
```
**Severity**
I think this is 'annoyance' level. I'd like to train colleagues on how to use joins, and this makes it very confusing to explain.

**Additional context**
Generated SQL:
```
SELECT "Account"."email" AS "email", count(*) AS "count"
FROM "public"."subscription_history" INNER JOIN "public"."users_plus" "Account" ON "public"."subscription_history"."account_id" = "Account"."account_id"
WHERE "public"."subscription_history"."plus_video" = TRUE
GROUP BY "Account"."email"
ORDER BY "count" DESC, "Account"."email" ASC
```
Highlight: **"users_plus" "Account"**

Thank you for your work on Metabase. It has transformed our growing company's ability to democratize data access & visibility.

**Comment by flamber** [comment]
Hi @rigoleto 
I guess it takes the normalized column name (not Data Model name) from the FK column.
Not sure if that is intentional to avoid conflicting names, when joining multiple tables to the same.

Reproduce with Sample Data on `master`:
1. Custom question > Sample Dataset > Orders
2. Join with Reviews on Product_ID=Product_ID
3. Click "Pick a column to group by", and it shows the joined table as **Product**

**Comment by camsaul** [comment]
I do not know why we'd be using the FK column name and not the actual table name for the join. Seems silly. Should be straightforward to fix

**Comment by camsaul** [comment]
It looks like this is determined here: 

https://github.com/metabase/metabase/blob/ab22e99c86062208451c450220ff2c94f6b0f3b9/frontend/src/metabase-lib/lib/DimensionOptions.js#L53-L57

Seeing if I can come up with a quick fix

**Comment by camsaul** [comment]
Fixed by #13503
